app/src/main/java/org/greenstand/android/TreeTracker/utilities/ImageUtils.kt

```diff
@@ -297,38 +296,134 @@ object ImageUtils {
             }
         }
 
-        var sum = 0;
+        var sum = 0
         for (row in 0 until rows) {
             for (col in 0 until cols) {
-                val horizontalGradientAtRowColumn = horizontalGradients[row][col];
-                val verticalGradientAtRowColumn = verticalGradients[row][col];
-                if (kotlin.math.abs(horizontalGradientAtRowColumn) > kotlin.math.abs(verticalGradientAtRowColumn)) {
-                    sum += horizontalGradientAtRowColumn * horizontalGradientAtRowColumn
+                val horizontalGradientAtRowColumn = horizontalGradients[row][col]
+                val verticalGradientAtRowColumn = verticalGradients[row][col]
+                sum += if (abs(horizontalGradientAtRowColumn) > abs(verticalGradientAtRowColumn)) {
+                    horizontalGradientAtRowColumn * horizontalGradientAtRowColumn
                 } else {
-                    sum += verticalGradientAtRowColumn * verticalGradientAtRowColumn
+                    verticalGradientAtRowColumn * verticalGradientAtRowColumn
                 }
             }
         }
-        return sum / (rows * cols).toDouble();
+        return sum / (rows * cols).toDouble()
     }
 
+    fun resizedImage(path: String): Bitmap {
+
+        /* There isn't enough memory to open up more than a couple camera photos */
+        /* So pre-scale the target bitmap into which the file is decoded */
+
+        /* Get the size of the image */
+        val bmOptions = BitmapFactory.Options()
+        bmOptions.inJustDecodeBounds = true
+        BitmapFactory.decodeFile(path, bmOptions)
+        val imageHeight = bmOptions.outHeight
+        val imageWidth = bmOptions.outWidth
+
+        // Calculate your sampleSize based on the requiredWidth and
+        // originalWidth
+        // For e.g you want the width to stay consistent at 500dp
+        var requiredWidth = 800
+
+        if (imageHeight > imageWidth) {
+            requiredWidth = 600
+        }
+
+        var sampleSize = ceil((imageWidth.toFloat() / requiredWidth.toFloat()).toDouble()).toInt()
+
+        // If the original image is smaller than required, don't sample
+        if (sampleSize < 1) {
+            sampleSize = 1
+        }
+
+        bmOptions.inSampleSize = sampleSize
+        bmOptions.inPurgeable = true
+        bmOptions.inPreferredConfig = Bitmap.Config.RGB_565
+        bmOptions.inJustDecodeBounds = false
+
+        /* Decode the JPEG file into a Bitmap */
+        val bitmap = BitmapFactory.decodeFile(path, bmOptions)
+
+        val rotatedBitmap = Bitmap.createBitmap(bitmap, 0, 0,
+            bmOptions.outWidth, bmOptions.outHeight, Matrix(), true)
+
+        val compressionQuality = 100
+        val byteArrayBitmapStream = ByteArrayOutputStream()
+        rotatedBitmap.compress(Bitmap.CompressFormat.PNG, compressionQuality,
+            byteArrayBitmapStream)
+
+        return rotatedBitmap
+    }
+
+    fun base64Image(path: String): String {
```

Comment:
I see this function is not used anywhere, why do we need it?  Also, if there is a need, can't this function just call `resizedImage` and then proceed to call `Base64.encodeToString()` instead of implementing the same code as `resizedImage()` does? Or am I missing something nuance.